src/semaphore.c

```diff
@@ -716,6 +771,11 @@ _dispatch_thread_semaphore_create(void)
 	int ret = sem_init(s4, 0, 0);
 	DISPATCH_SEMAPHORE_VERIFY_RET(ret);
 	return (_dispatch_thread_semaphore_t)s4;
+#elif USE_FUTEX_SEM
+	dispatch_futex_t *futex = malloc(sizeof(*futex));
```

Comment:
> Only other thing I can think off is to use as you say above and return the address
of the thread local variable in the allocation ? Is that's what you meant?

yes